### ISSUE TITLE
Unexpected argument 'idpb' in algorithms.varAnd

### ISSUE THREAD

**Issue by davinellulinvega** [DEAP/deap#242]
Hello everyone,

First and foremost, since I am new to DEAP and it is my first time submitting an issue here, I would like to thank you for creating such a wonderful and easy to use library.
Now back to business. For my PhD project I am using DEAP (v1.0.2-1) to coevolve neural network's weights to teach a robot to survive. Individuals within each specie are designed as matrices and I am using Variation algorithm to perform the crossover and mutate part.
I closely followed the **coop_base.py** and **coop_gen.py** examples to write my main function. However, upon running the code, I get the following error:
`Traceback (most recent call last):
  File "init.py", line 77, in <module>
    Genetic.main()
  File "/path/to/my/project/Genetic.py", line 155, in main
    s = algorithms.varAnd(s, toolbox, 0.6, 0.1) 
  File "/usr/lib/python3.6/site-packages/deap/algorithms.py", line 77, in varAnd
    offspring[i], = toolbox.mutate(offspring[i])
TypeError: mutGaussian() got an unexpected keyword argument 'idpb'
`
At line 155 in **Genetic.py**, **s** contains all individuals from one specie.
Did I do something wrong here, to get this error? Or is this a bug in the framework?
If you need more information to solve this problem please let me know I will be glad to provide any specifics.

Thank you very much in advance for your help.
Sincerely.

**Comment by davinellulinvega** [comment]
Well noticing that I did not have the latest version of DEAP on my computer I upgraded from the 1.0.2-1 to the 1.2.2. And I am (not so) glad to report that I have the same issue on this version as well.

Sincerely.

**Comment by fmder** [comment]
There is a typo. It is indpb, for independent probability, not idpb.

Le 13 nov. 2017 18:11, "davinellulinvega" <notifications@github.com> a
écrit :

Well noticing that I did not have the latest version of DEAP on my computer
I upgraded from the 1.0.2-1 to the 1.2.2. And I am (not so) glad to report
that I have the same issue on this version as well.

Sincerely.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/DEAP/deap/issues/242#issuecomment-344090714>, or mute
the thread
